driver/src/orderbook/streamed/state.rs

```diff
@@ -0,0 +1,561 @@
+use super::*;
+use crate::contracts::{
+    stablex_auction_element,
+    stablex_contract::batch_exchange::{event_data::*, Event},
+};
+use crate::models::Order as ModelOrder;
+use ethcontract::U256;
+use serde::{Deserialize, Serialize};
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::fmt;
+use std::iter::Iterator;
+
+// TODO: Should we handle https://github.com/gnosis/dex-contracts/issues/620 ?
+// There is a workaround detailed in the issue it hasn't been implemented.
+
+// Most types, fields, functions in this module mirror the smart contract because we need to
+// emulate what it does based on the events it emits.
+
+/// The orderbook state as built from received events.
+///
+/// Note that there is no way to revert an event. The order in which events are received matters
+/// Applying events `A B` does not always result in the same state as `B A`. For example, a Trade
+/// can only be processed if the OrderPlacement for the traded order has previously been observed.
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
```

Comment:
It makes sense because the fee token emits a TokenListing event like all the other tokens which means it will be included in the history of all events.